API connector: can't create connector from yaml file

UI wizard still can't create API connector from yaml either from URL or direct upload. There's an exception in `syndesis-rest` log.

Cc @zregvart 

<details>
  <summary>syndesis-rest log</summary>

```
2018-01-10 07:14:32.040 ERROR [-,53a4d55d3b93cdcc,53a4d55d3b93cdcc,false] 1 --- [  XNIO-3 task-5] .s.r.v.h.e.SyndesisServerExceptionMapper : Internal Server Exception. Unable to process JSON references

java.lang.IllegalStateException: Unable to process JSON references
	at io.syndesis.connector.generator.swagger.JsonSchemaHelper.resolveSchemaForReference(JsonSchemaHelper.java:105) ~[connector-generator-1.2-SNAPSHOT.jar!/:1.2-SNAPSHOT]
	at io.syndesis.connector.generator.swagger.DataShapeHelper.createShapeFromReference(DataShapeHelper.java:93) ~[connector-generator-1.2-SNAPSHOT.jar!/:1.2-SNAPSHOT]
	at io.syndesis.connector.generator.swagger.DataShapeHelper.createShapeFromProperty(DataShapeHelper.java:89) ~[connector-generator-1.2-SNAPSHOT.jar!/:1.2-SNAPSHOT]
	at io.syndesis.connector.generator.swagger.DataShapeHelper.createShapeFromResponse(DataShapeHelper.java:59) ~[connector-generator-1.2-SNAPSHOT.jar!/:1.2-SNAPSHOT]
	at io.syndesis.connector.generator.swagger.SwaggerUnifiedShapeConnectorGenerator.lambda$createDescriptor$4(SwaggerUnifiedShapeConnectorGenerator.java:166) ~[connector-generator-1.2-SNAPSHOT.jar!/:1.2-SNAPSHOT]
	at java.util.Optional.map(Optional.java:215) ~[na:1.8.0_131]
	at io.syndesis.connector.generator.swagger.SwaggerUnifiedShapeConnectorGenerator.createDescriptor(SwaggerUnifiedShapeConnectorGenerator.java:166) ~[connector-generator-1.2-SNAPSHOT.jar!/:1.2-SNAPSHOT]
	at io.syndesis.connector.generator.swagger.BaseSwaggerConnectorGenerator.configureConnector(BaseSwaggerConnectorGenerator.java:191) ~[connector-generator-1.2-SNAPSHOT.jar!/:1.2-SNAPSHOT]
	at io.syndesis.connector.generator.swagger.BaseSwaggerConnectorGenerator.generate(BaseSwaggerConnectorGenerator.java:86) ~[connector-generator-1.2-SNAPSHOT.jar!/:1.2-SNAPSHOT]
	at io.syndesis.rest.v1.handler.connection.CustomConnectorHandler.lambda$create$0(CustomConnectorHandler.java:64) ~[rest-1.2-SNAPSHOT.jar!/:1.2-SNAPSHOT]
	at io.syndesis.rest.v1.handler.connection.BaseConnectorGeneratorHandler.withGeneratorAndTemplate(BaseConnectorGeneratorHandler.java:52) ~[rest-1.2-SNAPSHOT.jar!/:1.2-SNAPSHOT]
	at io.syndesis.rest.v1.handler.connection.CustomConnectorHandler.create(CustomConnectorHandler.java:63) ~[rest-1.2-SNAPSHOT.jar!/:1.2-SNAPSHOT]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:1.8.0_131]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[na:1.8.0_131]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:1.8.0_131]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[na:1.8.0_131]
	at org.jboss.resteasy.core.MethodInjectorImpl.invoke(MethodInjectorImpl.java:140) ~[resteasy-jaxrs-3.1.4.Final.jar!/:3.1.4.Final]
	at org.jboss.resteasy.core.ResourceMethodInvoker.invokeOnTarget(ResourceMethodInvoker.java:294) ~[resteasy-jaxrs-3.1.4.Final.jar!/:3.1.4.Final]
	at org.jboss.resteasy.core.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:248) ~[resteasy-jaxrs-3.1.4.Final.jar!/:3.1.4.Final]
	at org.jboss.resteasy.core.ResourceLocatorInvoker.invokeOnTargetObject(ResourceLocatorInvoker.java:138) ~[resteasy-jaxrs-3.1.4.Final.jar!/:3.1.4.Final]
	at org.jboss.resteasy.core.ResourceLocatorInvoker.invoke(ResourceLocatorInvoker.java:101) ~[resteasy-jaxrs-3.1.4.Final.jar!/:3.1.4.Final]
	at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:402) [resteasy-jaxrs-3.1.4.Final.jar!/:3.1.4.Final]
	at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:209) [resteasy-jaxrs-3.1.4.Final.jar!/:3.1.4.Final]
	at org.jboss.resteasy.plugins.server.servlet.ServletContainerDispatcher.service(ServletContainerDispatcher.java:227) [resteasy-jaxrs-3.1.4.Final.jar!/:3.1.4.Final]
	at org.jboss.resteasy.plugins.server.servlet.HttpServletDispatcher.service(HttpServletDispatcher.java:56) [resteasy-jaxrs-3.1.4.Final.jar!/:3.1.4.Final]
	at org.jboss.resteasy.plugins.server.servlet.HttpServletDispatcher.service(HttpServletDispatcher.java:51) [resteasy-jaxrs-3.1.4.Final.jar!/:3.1.4.Final]
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:790) [javax.servlet-api-3.1.0.jar!/:3.1.0]
	at io.undertow.servlet.handlers.ServletHandler.handleRequest(ServletHandler.java:85) [undertow-servlet-1.4.21.Final.jar!/:1.4.21.Final]
	at io.undertow.servlet.handlers.FilterHandler$FilterChainImpl.doFilter(FilterHandler.java:129) [undertow-servlet-1.4.21.Final.jar!/:1.4.21.Final]
	at org.springframework.boot.web.filter.ApplicationContextHeaderFilter.doFilterInternal(ApplicationContextHeaderFilter.java:55) [spring-boot-1.5.8.RELEASE.jar!/:1.5.8.RELEASE]
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107) [spring-web-4.3.12.RELEASE.jar!/:4.3.12.RELEASE]
	at io.undertow.servlet.core.ManagedFilter.doFilter(ManagedFilter.java:61) [undertow-servlet-1.4.21.Final.jar!/:1.4.21.Final]
	at io.undertow.servlet.handlers.FilterHandler$FilterChainImpl.doFilter(FilterHandler.java:131) [undertow-servlet-1.4.21.Final.jar!/:1.4.21.Final]
	at org.springframework.web.filter.CorsFilter.doFilterInternal(CorsFilter.java:96) [spring-web-4.3.12.RELEASE.jar!/:4.3.12.RELEASE]
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107) [spring-web-4.3.12.RELEASE.jar!/:4.3.12.RELEASE]
	at io.undertow.servlet.core.ManagedFilter.doFilter(ManagedFilter.java:61) [undertow-servlet-1.4.21.Final.jar!/:1.4.21.Final]
	at io.undertow.servlet.handlers.FilterHandler$FilterChainImpl.doFilter(FilterHandler.java:131) [undertow-servlet-1.4.21.Final.jar!/:1.4.21.Final]
	at org.springframework.boot.actuate.trace.WebRequestTraceFilter.doFilterInternal(WebRequestTraceFilter.java:110) [spring-boot-actuator-1.5.8.RELEASE.jar!/:1.5.8.RELEASE]
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107) [spring-web-4.3.12.RELEASE.jar!/:4.3.12.RELEASE]
	at io.undertow.servlet.core.ManagedFilter.doFilter(ManagedFilter.java:61) [undertow-servlet-1.4.21.Final.jar!/:1.4.21.Final]
	at io.undertow.servlet.handlers.FilterHandler$FilterChainImpl.doFilter(FilterHandler.java:131) [undertow-servlet-1.4.21.Final.jar!/:1.4.21.Final]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:317) [spring-security-web-4.2.3.RELEASE.jar!/:4.2.3.RELEASE]
	at org.springframework.security.web.access.intercept.FilterSecurityInterceptor.invoke(FilterSecurityInterceptor.java:127) [spring-security-web-4.2.3.RELEASE.jar!/:4.2.3.RELEASE]
	at org.springframework.security.web.access.intercept.FilterSecurityInterceptor.doFilter(FilterSecurityInterceptor.java:91) [spring-security-web-4.2.3.RELEASE.jar!/:4.2.3.RELEASE]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331) [spring-security-web-4.2.3.RELEASE.jar!/:4.2.3.RELEASE]
	at org.springframework.security.web.access.ExceptionTranslationFilter.doFilter(ExceptionTranslationFilter.java:114) [spring-security-web-4.2.3.RELEASE.jar!/:4.2.3.RELEASE]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331) [spring-security-web-4.2.3.RELEASE.jar!/:4.2.3.RELEASE]
	at org.springframework.security.web.session.SessionManagementFilter.doFilter(SessionManagementFilter.java:137) [spring-security-web-4.2.3.RELEASE.jar!/:4.2.3.RELEASE]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331) [spring-security-web-4.2.3.RELEASE.jar!/:4.2.3.RELEASE]
	at org.springframework.security.web.authentication.AnonymousAuthenticationFilter.doFilter(AnonymousAuthenticationFilter.java:111) [spring-security-web-4.2.3.RELEASE.jar!/:4.2.3.RELEASE]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331) [spring-security-web-4.2.3.RELEASE.jar!/:4.2.3.RELEASE]
	at org.springframework.security.web.authentication.AnonymousAuthenticationFilter.doFilter(AnonymousAuthenticationFilter.java:111) [spring-security-web-4.2.3.RELEASE.jar!/:4.2.3.RELEASE]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331) [spring-security-web-4.2.3.RELEASE.jar!/:4.2.3.RELEASE]
	at org.springframework.security.web.servletapi.SecurityContextHolderAwareRequestFilter.doFilter(SecurityContextHolderAwareRequestFilter.java:170) [spring-security-web-4.2.3.RELEASE.jar!/:4.2.3.RELEASE]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331) [spring-security-web-4.2.3.RELEASE.jar!/:4.2.3.RELEASE]
	at org.springframework.security.web.savedrequest.RequestCacheAwareFilter.doFilter(RequestCacheAwareFilter.java:63) [spring-security-web-4.2.3.RELEASE.jar!/:4.2.3.RELEASE]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331) [spring-security-web-4.2.3.RELEASE.jar!/:4.2.3.RELEASE]
	at org.springframework.security.web.authentication.preauth.AbstractPreAuthenticatedProcessingFilter.doFilter(AbstractPreAuthenticatedProcessingFilter.java:121) [spring-security-web-4.2.3.RELEASE.jar!/:4.2.3.RELEASE]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331) [spring-security-web-4.2.3.RELEASE.jar!/:4.2.3.RELEASE]
	at org.springframework.security.web.authentication.logout.LogoutFilter.doFilter(LogoutFilter.java:116) [spring-security-web-4.2.3.RELEASE.jar!/:4.2.3.RELEASE]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331) [spring-security-web-4.2.3.RELEASE.jar!/:4.2.3.RELEASE]
	at org.springframework.security.web.header.HeaderWriterFilter.doFilterInternal(HeaderWriterFilter.java:64) [spring-security-web-4.2.3.RELEASE.jar!/:4.2.3.RELEASE]
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107) [spring-web-4.3.12.RELEASE.jar!/:4.3.12.RELEASE]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331) [spring-security-web-4.2.3.RELEASE.jar!/:4.2.3.RELEASE]
	at org.springframework.security.web.context.SecurityContextPersistenceFilter.doFilter(SecurityContextPersistenceFilter.java:105) [spring-security-web-4.2.3.RELEASE.jar!/:4.2.3.RELEASE]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331) [spring-security-web-4.2.3.RELEASE.jar!/:4.2.3.RELEASE]
	at org.springframework.security.web.context.request.async.WebAsyncManagerIntegrationFilter.doFilterInternal(WebAsyncManagerIntegrationFilter.java:56) [spring-security-web-4.2.3.RELEASE.jar!/:4.2.3.RELEASE]
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107) [spring-web-4.3.12.RELEASE.jar!/:4.3.12.RELEASE]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331) [spring-security-web-4.2.3.RELEASE.jar!/:4.2.3.RELEASE]
	at org.springframework.security.web.FilterChainProxy.doFilterInternal(FilterChainProxy.java:214) [spring-security-web-4.2.3.RELEASE.jar!/:4.2.3.RELEASE]
	at org.springframework.security.web.FilterChainProxy.doFilter(FilterChainProxy.java:177) [spring-security-web-4.2.3.RELEASE.jar!/:4.2.3.RELEASE]
	at org.springframework.web.filter.DelegatingFilterProxy.invokeDelegate(DelegatingFilterProxy.java:347) [spring-web-4.3.12.RELEASE.jar!/:4.3.12.RELEASE]
	at org.springframework.web.filter.DelegatingFilterProxy.doFilter(DelegatingFilterProxy.java:263) [spring-web-4.3.12.RELEASE.jar!/:4.3.12.RELEASE]
	at io.undertow.servlet.core.ManagedFilter.doFilter(ManagedFilter.java:61) [undertow-servlet-1.4.21.Final.jar!/:1.4.21.Final]
	at io.undertow.servlet.handlers.FilterHandler$FilterChainImpl.doFilter(FilterHandler.java:131) [undertow-servlet-1.4.21.Final.jar!/:1.4.21.Final]
	at org.springframework.web.filter.RequestContextFilter.doFilterInternal(RequestContextFilter.java:99) [spring-web-4.3.12.RELEASE.jar!/:4.3.12.RELEASE]
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107) [spring-web-4.3.12.RELEASE.jar!/:4.3.12.RELEASE]
	at io.undertow.servlet.core.ManagedFilter.doFilter(ManagedFilter.java:61) [undertow-servlet-1.4.21.Final.jar!/:1.4.21.Final]
	at io.undertow.servlet.handlers.FilterHandler$FilterChainImpl.doFilter(FilterHandler.java:131) [undertow-servlet-1.4.21.Final.jar!/:1.4.21.Final]
	at org.springframework.web.filter.HttpPutFormContentFilter.doFilterInternal(HttpPutFormContentFilter.java:108) [spring-web-4.3.12.RELEASE.jar!/:4.3.12.RELEASE]
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107) [spring-web-4.3.12.RELEASE.jar!/:4.3.12.RELEASE]
	at io.undertow.servlet.core.ManagedFilter.doFilter(ManagedFilter.java:61) [undertow-servlet-1.4.21.Final.jar!/:1.4.21.Final]
	at io.undertow.servlet.handlers.FilterHandler$FilterChainImpl.doFilter(FilterHandler.java:131) [undertow-servlet-1.4.21.Final.jar!/:1.4.21.Final]
	at org.springframework.web.filter.HiddenHttpMethodFilter.doFilterInternal(HiddenHttpMethodFilter.java:81) [spring-web-4.3.12.RELEASE.jar!/:4.3.12.RELEASE]
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107) [spring-web-4.3.12.RELEASE.jar!/:4.3.12.RELEASE]
	at io.undertow.servlet.core.ManagedFilter.doFilter(ManagedFilter.java:61) [undertow-servlet-1.4.21.Final.jar!/:1.4.21.Final]
	at io.undertow.servlet.handlers.FilterHandler$FilterChainImpl.doFilter(FilterHandler.java:131) [undertow-servlet-1.4.21.Final.jar!/:1.4.21.Final]
	at org.springframework.cloud.sleuth.instrument.web.TraceFilter.doFilter(TraceFilter.java:186) [spring-cloud-sleuth-core-1.2.5.RELEASE.jar!/:1.2.5.RELEASE]
	at io.undertow.servlet.core.ManagedFilter.doFilter(ManagedFilter.java:61) [undertow-servlet-1.4.21.Final.jar!/:1.4.21.Final]
	at io.undertow.servlet.handlers.FilterHandler$FilterChainImpl.doFilter(FilterHandler.java:131) [undertow-servlet-1.4.21.Final.jar!/:1.4.21.Final]
	at org.springframework.web.filter.CharacterEncodingFilter.doFilterInternal(CharacterEncodingFilter.java:197) [spring-web-4.3.12.RELEASE.jar!/:4.3.12.RELEASE]
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107) [spring-web-4.3.12.RELEASE.jar!/:4.3.12.RELEASE]
	at io.undertow.servlet.core.ManagedFilter.doFilter(ManagedFilter.java:61) [undertow-servlet-1.4.21.Final.jar!/:1.4.21.Final]
	at io.undertow.servlet.handlers.FilterHandler$FilterChainImpl.doFilter(FilterHandler.java:131) [undertow-servlet-1.4.21.Final.jar!/:1.4.21.Final]
	at org.springframework.boot.actuate.autoconfigure.MetricsFilter.doFilterInternal(MetricsFilter.java:106) [spring-boot-actuator-1.5.8.RELEASE.jar!/:1.5.8.RELEASE]
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107) [spring-web-4.3.12.RELEASE.jar!/:4.3.12.RELEASE]
	at io.undertow.servlet.core.ManagedFilter.doFilter(ManagedFilter.java:61) [undertow-servlet-1.4.21.Final.jar!/:1.4.21.Final]
	at io.undertow.servlet.handlers.FilterHandler$FilterChainImpl.doFilter(FilterHandler.java:131) [undertow-servlet-1.4.21.Final.jar!/:1.4.21.Final]
	at io.undertow.servlet.handlers.FilterHandler.handleRequest(FilterHandler.java:84) [undertow-servlet-1.4.21.Final.jar!/:1.4.21.Final]
	at io.undertow.servlet.handlers.security.ServletSecurityRoleHandler.handleRequest(ServletSecurityRoleHandler.java:62) [undertow-servlet-1.4.21.Final.jar!/:1.4.21.Final]
	at io.undertow.servlet.handlers.ServletChain$1.handleRequest(ServletChain.java:64) [undertow-servlet-1.4.21.Final.jar!/:1.4.21.Final]
	at io.undertow.servlet.handlers.ServletDispatchingHandler.handleRequest(ServletDispatchingHandler.java:36) [undertow-servlet-1.4.21.Final.jar!/:1.4.21.Final]
	at io.undertow.servlet.handlers.security.SSLInformationAssociationHandler.handleRequest(SSLInformationAssociationHandler.java:131) [undertow-servlet-1.4.21.Final.jar!/:1.4.21.Final]
	at io.undertow.servlet.handlers.security.ServletAuthenticationCallHandler.handleRequest(ServletAuthenticationCallHandler.java:57) [undertow-servlet-1.4.21.Final.jar!/:1.4.21.Final]
	at io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43) [undertow-core-1.4.21.Final.jar!/:1.4.21.Final]
	at io.undertow.security.handlers.AbstractConfidentialityHandler.handleRequest(AbstractConfidentialityHandler.java:46) [undertow-core-1.4.21.Final.jar!/:1.4.21.Final]
	at io.undertow.servlet.handlers.security.ServletConfidentialityConstraintHandler.handleRequest(ServletConfidentialityConstraintHandler.java:64) [undertow-servlet-1.4.21.Final.jar!/:1.4.21.Final]
	at io.undertow.security.handlers.AuthenticationMechanismsHandler.handleRequest(AuthenticationMechanismsHandler.java:60) [undertow-core-1.4.21.Final.jar!/:1.4.21.Final]
	at io.undertow.servlet.handlers.security.CachedAuthenticatedSessionHandler.handleRequest(CachedAuthenticatedSessionHandler.java:77) [undertow-servlet-1.4.21.Final.jar!/:1.4.21.Final]
	at io.undertow.security.handlers.AbstractSecurityContextAssociationHandler.handleRequest(AbstractSecurityContextAssociationHandler.java:43) [undertow-core-1.4.21.Final.jar!/:1.4.21.Final]
	at io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43) [undertow-core-1.4.21.Final.jar!/:1.4.21.Final]
	at io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43) [undertow-core-1.4.21.Final.jar!/:1.4.21.Final]
	at io.undertow.servlet.handlers.ServletInitialHandler.handleFirstRequest(ServletInitialHandler.java:292) [undertow-servlet-1.4.21.Final.jar!/:1.4.21.Final]
	at io.undertow.servlet.handlers.ServletInitialHandler.access$100(ServletInitialHandler.java:81) [undertow-servlet-1.4.21.Final.jar!/:1.4.21.Final]
	at io.undertow.servlet.handlers.ServletInitialHandler$2.call(ServletInitialHandler.java:138) [undertow-servlet-1.4.21.Final.jar!/:1.4.21.Final]
	at io.undertow.servlet.handlers.ServletInitialHandler$2.call(ServletInitialHandler.java:135) [undertow-servlet-1.4.21.Final.jar!/:1.4.21.Final]
	at io.undertow.servlet.core.ServletRequestContextThreadSetupAction$1.call(ServletRequestContextThreadSetupAction.java:48) [undertow-servlet-1.4.21.Final.jar!/:1.4.21.Final]
	at io.undertow.servlet.core.ContextClassLoaderSetupAction$1.call(ContextClassLoaderSetupAction.java:43) [undertow-servlet-1.4.21.Final.jar!/:1.4.21.Final]
	at io.undertow.servlet.handlers.ServletInitialHandler.dispatchRequest(ServletInitialHandler.java:272) [undertow-servlet-1.4.21.Final.jar!/:1.4.21.Final]
	at io.undertow.servlet.handlers.ServletInitialHandler.access$000(ServletInitialHandler.java:81) [undertow-servlet-1.4.21.Final.jar!/:1.4.21.Final]
	at io.undertow.servlet.handlers.ServletInitialHandler$1.handleRequest(ServletInitialHandler.java:104) [undertow-servlet-1.4.21.Final.jar!/:1.4.21.Final]
	at io.undertow.server.Connectors.executeRootHandler(Connectors.java:332) [undertow-core-1.4.21.Final.jar!/:1.4.21.Final]
	at io.undertow.server.HttpServerExchange$1.run(HttpServerExchange.java:830) [undertow-core-1.4.21.Final.jar!/:1.4.21.Final]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) [na:1.8.0_131]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) [na:1.8.0_131]
	at java.lang.Thread.run(Thread.java:748) [na:1.8.0_131]
Caused by: com.fasterxml.jackson.core.JsonParseException: Unexpected character ('-' (code 45)) in numeric value: expected digit (0-9) to follow minus sign, for valid numeric value
 at [Source: mem:specification; line: 1, column: 3]
	at com.fasterxml.jackson.core.JsonParser._constructError(JsonParser.java:1702) ~[jackson-core-2.8.10.jar!/:2.8.10]
	at com.fasterxml.jackson.core.base.ParserMinimalBase._reportError(ParserMinimalBase.java:558) ~[jackson-core-2.8.10.jar!/:2.8.10]
	at com.fasterxml.jackson.core.base.ParserBase.reportUnexpectedNumberChar(ParserBase.java:1058) ~[jackson-core-2.8.10.jar!/:2.8.10]
	at com.fasterxml.jackson.core.json.UTF8StreamJsonParser._handleInvalidNumberStart(UTF8StreamJsonParser.java:2811) ~[jackson-core-2.8.10.jar!/:2.8.10]
	at com.fasterxml.jackson.core.json.UTF8StreamJsonParser._parseNegNumber(UTF8StreamJsonParser.java:1446) ~[jackson-core-2.8.10.jar!/:2.8.10]
	at com.fasterxml.jackson.core.json.UTF8StreamJsonParser._nextTokenNotInObject(UTF8StreamJsonParser.java:861) ~[jackson-core-2.8.10.jar!/:2.8.10]
	at com.fasterxml.jackson.core.json.UTF8StreamJsonParser.nextToken(UTF8StreamJsonParser.java:772) ~[jackson-core-2.8.10.jar!/:2.8.10]
	at com.fasterxml.jackson.databind.ObjectMapper._initForReading(ObjectMapper.java:3850) ~[jackson-databind-2.8.10.jar!/:2.8.10]
	at com.fasterxml.jackson.databind.ObjectMapper._readMapAndClose(ObjectMapper.java:3799) ~[jackson-databind-2.8.10.jar!/:2.8.10]
	at com.fasterxml.jackson.databind.ObjectMapper.readTree(ObjectMapper.java:2474) ~[jackson-databind-2.8.10.jar!/:2.8.10]
	at me.andrz.jackson.JsonContext.getNode(JsonContext.java:51) ~[jackson-json-reference-core-0.2.1.jar!/:na]
	at me.andrz.jackson.JsonReferenceProcessor.process(JsonReferenceProcessor.java:107) ~[jackson-json-reference-core-0.2.1.jar!/:na]
	at me.andrz.jackson.JsonReferenceProcessor.process(JsonReferenceProcessor.java:96) ~[jackson-json-reference-core-0.2.1.jar!/:na]
	at io.syndesis.connector.generator.swagger.JsonSchemaHelper.resolveSchemaForReference(JsonSchemaHelper.java:103) ~[connector-generator-1.2-SNAPSHOT.jar!/:1.2-SNAPSHOT]
	... 125 common frames omitted
```

</details>

--- a/app/rest/connector-generator/src/main/java/io/syndesis/connector/generator/swagger/SwaggerHelper.java
+++ b/app/rest/connector-generator/src/main/java/io/syndesis/connector/generator/swagger/SwaggerHelper.java
@@ -66,6 +66,16 @@ public final class SwaggerHelper {
         // utility class
     }
 
+    /* default */ static JsonNode convertToJson(final String specification) throws IOException, JsonProcessingException {
+        final JsonNode specRoot;
+        if (specification.matches("\\s+\\{")) {
+            specRoot = JSON_MAPPER.readTree(specification);
+        } else {
+            specRoot = JSON_MAPPER.convertValue(YAML_PARSER.load(specification), JsonNode.class);
+        }
+        return specRoot;
+    }
+
     /* default */ static SwaggerModelInfo parse(final String specification, final boolean validate) {
         final SwaggerModelInfo.Builder resultBuilder = new SwaggerModelInfo.Builder();
 
@@ -99,11 +109,16 @@ public final class SwaggerHelper {
     }
 
     /* default */ static String resolve(final String specification) throws Exception {
+        final String specificationToUse;
         if (specification.toLowerCase().startsWith("http")) {
-            return RemoteUrl.urlToString(specification, null);
+            specificationToUse = RemoteUrl.urlToString(specification, null);
+        } else {
+            specificationToUse = specification;
         }
 
-        return specification;
+        final JsonNode node = convertToJson(specificationToUse);
+
+        return Json.mapper().writeValueAsString(node);
     }
 
     /* default */ static String serialize(final Swagger swagger) {
@@ -137,12 +152,7 @@ public final class SwaggerHelper {
 
     private static SwaggerModelInfo validateJSonSchema(final String specification, final Swagger model) {
         try {
-            final JsonNode specRoot;
-            if (specification.matches("\\s+\\{")) {
-                specRoot = JSON_MAPPER.readTree(specification);
-            } else {
-                specRoot = JSON_MAPPER.convertValue(YAML_PARSER.load(specification), JsonNode.class);
-            }
+            final JsonNode specRoot = convertToJson(specification);
             final ProcessingReport report = SWAGGER_2_0_SCHEMA.validate(specRoot);
             final List<Violation> errors = new ArrayList<>();
             final List<Violation> warnings = new ArrayList<>();

--- a/app/rest/connector-generator/src/test/java/io/syndesis/connector/generator/swagger/BaseSwaggerGeneratorExampleTest.java
+++ b/app/rest/connector-generator/src/test/java/io/syndesis/connector/generator/swagger/BaseSwaggerGeneratorExampleTest.java
@@ -52,6 +52,8 @@ abstract class BaseSwaggerGeneratorExampleTest extends AbstractSwaggerConnectorT
 
         final Connector generated = generator().generate(SWAGGER_TEMPLATE, connectorSettings);
 
+        Json.mapper().writeValue(System.out, generated);
+
         final Map<String, String> generatedConfiguredProperties = generated.getConfiguredProperties();
         final String generatedSpecification = generatedConfiguredProperties.get("specification");
 

--- a/app/rest/connector-generator/src/test/java/io/syndesis/connector/generator/swagger/SwaggerUnifiedShapeGeneratorExampleTests.java
+++ b/app/rest/connector-generator/src/test/java/io/syndesis/connector/generator/swagger/SwaggerUnifiedShapeGeneratorExampleTests.java
@@ -45,7 +45,7 @@ public class SwaggerUnifiedShapeGeneratorExampleTests extends BaseSwaggerGenerat
 
     @Parameters(name = "{0}")
     public static Iterable<String> parameters() {
-        return Arrays.asList("petstore", "basic_auth");
+        return Arrays.asList("petstore", "basic_auth", "todo");
     }
 
 }

--- a/app/rest/connector-generator/src/test/resources/swagger/todo.swagger.yaml
+++ b/app/rest/connector-generator/src/test/resources/swagger/todo.swagger.yaml
@@ -136,4 +136,5 @@ definitions:
         type: "integer"
         title: "Task completition status"
         description: "0 - ongoing, 1 - completed"
-        enum: [0, 1]
+        minimum: 0
+        maximum: 1

--- a/app/rest/connector-generator/src/test/resources/swagger/todo.unified_connector.json
+++ b/app/rest/connector-generator/src/test/resources/swagger/todo.unified_connector.json
@@ -1,0 +1,240 @@
+{
+  "name": "Todo App API",
+  "description": "unspecified",
+  "connectorGroup": {
+    "id": "swagger-connector-template"
+  },
+  "connectorGroupId": "swagger-connector-template",
+  "configuredProperties": {
+    "specification": "{\"swagger\": \"2.0\", \"info\": {\"title\": \"Todo App API\", \"version\": \"1.0.0\", \"license\": {\"name\": \"Apache 2.0\", \"url\": \"http://www.apache.org/licenses/LICENSE-2.0.html\"}}, \"host\": \"hostname\", \"schemes\": [ \"http\", \"https\" ], \"paths\": {\"/api\": {\"get\": {\"operationId\": \"operation-0\", \"tags\": [ \"tasks\", \"fetching\" ], \"summary\": \"List all tasks\", \"description\": \"Fetches all tasks from the database\", \"parameters\" : [ ], \"produces\": [ \"application/json\" ], \"responses\": {\"200\": {\"description\": \"All is good\", \"schema\": {\"type\": \"array\", \"items\": {\"$ref\": \"#/definitions/Task\"}}}}}, \"post\": {\"operationId\": \"operation-1\", \"tags\": [ \"tasks\", \"creating\" ], \"summary\": \"Create new task\", \"description\": \"Stores new task in the database\", \"produces\": [ \"application/json\" ], \"consumes\": [ \"application/json\" ], \"parameters\": [ {\"in\": \"body\", \"name\": \"body\", \"description\": \"Task to create\", \"required\": true, \"schema\": {\"$ref\": \"#/definitions/Task\"}} ], \"responses\": {\"201\": {\"description\": \"All is good\", \"schema\": {\"$ref\": \"#/definitions/Task\"}}}}}, \"/api/{id}\": {\"get\": {\"operationId\": \"operation-2\", \"tags\": [ \"tasks\", \"fetching\" ], \"summary\": \"Fetch task\", \"description\": \"Fetches task by given identifier\", \"produces\": [ \"application/json\" ], \"parameters\": [ {\"in\": \"path\", \"name\": \"id\", \"type\": \"integer\", \"format\": \"int64\", \"description\": \"Task identifier\", \"required\": true} ], \"responses\": {\"200\": {\"description\": \"All is good\", \"schema\": {\"$ref\": \"#/definitions/Task\"}}, \"404\": {\"description\": \"No task with provided identifier found\"}}}, \"put\": {\"operationId\": \"operation-3\", \"tags\": [ \"tasks\", \"updating\" ], \"summary\": \"Update task\", \"description\": \"Updates task by given identifier\", \"produces\": [ \"application/json\" ], \"consumes\": [ \"application/json\" ], \"parameters\": [ {\"in\": \"path\", \"name\": \"id\", \"type\": \"integer\", \"format\": \"int64\", \"description\": \"Task identifier\", \"required\": true}, {\"in\": \"body\", \"name\": \"body\", \"description\": \"Task with updates\", \"required\": true, \"schema\": {\"$ref\": \"#/definitions/Task\"}} ], \"responses\": {\"200\": {\"description\": \"All is good\", \"schema\": {\"$ref\": \"#/definitions/Task\"}}}}, \"delete\": {\"operationId\": \"operation-4\", \"tags\": [ \"tasks\", \"destruction\" ], \"summary\": \"Delete task\", \"description\": \"Deletes task by given identifier\", \"parameters\": [ {\"in\": \"path\", \"name\": \"id\", \"type\": \"integer\", \"format\": \"int64\", \"description\": \"Task identifier to delete\", \"required\": true} ], \"responses\": {\"204\": {\"description\": \"Task deleted\"}}}}}, \"securityDefinitions\": {\"username_password\": {\"type\": \"basic\"}}, \"definitions\": {\"Task\": {\"type\": \"object\", \"properties\": {\"id\": {\"type\": \"integer\", \"format\": \"int64\", \"title\": \"Task ID\", \"description\": \"Unique task identifier\"}, \"task\": {\"type\": \"string\", \"title\": \"The task\", \"description\": \"Task line\"}, \"completed\": {\"type\": \"integer\", \"title\": \"Task completition status\", \"description\": \"0 - ongoing, 1 - completed\", \"minimum\": 0, \"maximum\": 1}}}}}",
+    "authenticationType": "basic",
+    "host": "https://hostname"
+  },
+  "properties": {
+    "authenticationType": {
+      "kind": "property",
+      "displayName": "Authentication Type",
+      "group": "producer",
+      "label": "producer",
+      "required": false,
+      "type": "string",
+      "javaType": "java.lang.String",
+      "tags": [
+        "authentication-type"
+      ],
+      "deprecated": false,
+      "secret": false,
+      "componentProperty": true,
+      "defaultValue": "basic",
+      "enum": [
+        {
+          "value": "basic",
+          "label": "HTTP Basic Authentication"
+        }
+      ],
+      "description": "Type of authentication used to connect to the API"
+    },
+    "specification": {
+      "kind": "property",
+      "displayName": "Specification",
+      "group": "producer",
+      "label": "producer",
+      "required": false,
+      "type": "hidden",
+      "javaType": "java.lang.String",
+      "tags": [
+        "upload",
+        "url"
+      ],
+      "deprecated": false,
+      "secret": false,
+      "componentProperty": true,
+      "description": "Swagger specification of the service"
+    },
+    "basePath": {
+      "kind": "property",
+      "displayName": "Base path",
+      "group": "producer",
+      "label": "producer",
+      "required": false,
+      "type": "string",
+      "javaType": "java.lang.String",
+      "tags": [],
+      "deprecated": false,
+      "secret": false,
+      "componentProperty": true,
+      "description": "API basePath for example /v2. Default is unset if set overrides the value present in Swagger specification."
+    },
+    "host": {
+      "kind": "property",
+      "displayName": "Host",
+      "group": "producer",
+      "label": "producer",
+      "required": false,
+      "type": "string",
+      "javaType": "java.lang.String",
+      "tags": [],
+      "deprecated": false,
+      "secret": false,
+      "componentProperty": true,
+      "defaultValue": "https://hostname",
+      "description": "Scheme hostname and port to direct the HTTP requests to in the form of https://hostname:port. Can be configured at the endpoint component or in the correspoding REST configuration in the Camel Context. If you give this component a name (e.g. petstore) that REST configuration is consulted first rest-swagger next and global configuration last. If set overrides any value found in the Swagger specification RestConfiguration. Can be overriden in endpoint configuration."
+    },
+    "username": {
+      "kind": "property",
+      "displayName": "Username",
+      "group": "producer",
+      "label": "producer",
+      "required": true,
+      "type": "string",
+      "javaType": "java.lang.String",
+      "deprecated": false,
+      "secret": false,
+      "componentProperty": true,
+      "description": "Username to authenticate with"
+    },
+    "password": {
+      "kind": "property",
+      "displayName": "Password",
+      "group": "producer",
+      "label": "producer",
+      "required": true,
+      "type": "string",
+      "javaType": "java.lang.String",
+      "deprecated": false,
+      "secret": true,
+      "componentProperty": true,
+      "description": "Password to authenticate with"
+    }
+  },
+  "actions": [
+    {
+      "id": "io.syndesis:rest-swagger-connector:@syndesis-connectors.version@:_id_:operation-0",
+      "name": "List all tasks",
+      "description": "Fetches all tasks from the database",
+      "descriptor": {
+        "camelConnectorGAV": "io.syndesis:rest-swagger-connector:@syndesis-connectors.version@",
+        "camelConnectorPrefix": "swagger-operation",
+        "inputDataShape": {
+          "kind": "none"
+        },
+        "outputDataShape": {
+          "kind": "json-schema",
+          "specification": "{\"type\":\"object\",\"properties\":{\"id\":{\"type\":\"integer\",\"format\":\"int64\",\"title\":\"Task ID\",\"description\":\"Unique task identifier\"},\"task\":{\"type\":\"string\",\"title\":\"The task\",\"description\":\"Task line\"},\"completed\":{\"type\":\"integer\",\"title\":\"Task completition status\",\"description\":\"0 - ongoing, 1 - completed\",\"minimum\":0,\"maximum\":1}},\"$schema\":\"http://json-schema.org/schema#\",\"title\":\"Task\"}"
+        },
+        "configuredProperties": {
+          "operationId": "operation-0"
+        }
+      },
+      "tags": [
+        "fetching",
+        "tasks"
+      ],
+      "actionType": "connector",
+      "pattern": "To"
+    },
+    {
+      "id": "io.syndesis:rest-swagger-connector:@syndesis-connectors.version@:_id_:operation-1",
+      "name": "Create new task",
+      "description": "Stores new task in the database",
+      "descriptor": {
+        "camelConnectorGAV": "io.syndesis:rest-swagger-connector:@syndesis-connectors.version@",
+        "camelConnectorPrefix": "swagger-operation",
+        "inputDataShape": {
+          "kind": "json-schema",
+          "specification": "{\"$schema\":\"http://json-schema.org/draft-04/schema#\",\"type\":\"object\",\"properties\":{\"body\":{\"type\":\"object\",\"properties\":{\"id\":{\"type\":\"integer\",\"format\":\"int64\",\"title\":\"Task ID\",\"description\":\"Unique task identifier\"},\"task\":{\"type\":\"string\",\"title\":\"The task\",\"description\":\"Task line\"},\"completed\":{\"type\":\"integer\",\"title\":\"Task completition status\",\"description\":\"0 - ongoing, 1 - completed\",\"minimum\":0,\"maximum\":1}},\"$schema\":\"http://json-schema.org/schema#\",\"title\":\"Task\"}}}"
+        },
+        "outputDataShape": {
+          "kind": "json-schema",
+          "specification": "{\"type\":\"object\",\"properties\":{\"id\":{\"type\":\"integer\",\"format\":\"int64\",\"title\":\"Task ID\",\"description\":\"Unique task identifier\"},\"task\":{\"type\":\"string\",\"title\":\"The task\",\"description\":\"Task line\"},\"completed\":{\"type\":\"integer\",\"title\":\"Task completition status\",\"description\":\"0 - ongoing, 1 - completed\",\"minimum\":0,\"maximum\":1}},\"$schema\":\"http://json-schema.org/schema#\",\"title\":\"Task\"}"
+        },
+        "configuredProperties": {
+          "operationId": "operation-1"
+        }
+      },
+      "tags": [
+        "creating",
+        "tasks"
+      ],
+      "actionType": "connector",
+      "pattern": "To"
+    },
+    {
+      "id": "io.syndesis:rest-swagger-connector:@syndesis-connectors.version@:_id_:operation-2",
+      "name": "Fetch task",
+      "description": "Fetches task by given identifier",
+      "descriptor": {
+        "camelConnectorGAV": "io.syndesis:rest-swagger-connector:@syndesis-connectors.version@",
+        "camelConnectorPrefix": "swagger-operation",
+        "inputDataShape": {
+          "kind": "json-schema",
+          "specification": "{\"$schema\":\"http://json-schema.org/draft-04/schema#\",\"type\":\"object\",\"properties\":{\"parameters\":{\"type\":\"object\",\"properties\":{\"id\":{\"type\":\"integer\",\"title\":\"id\",\"description\":\"Task identifier\"}}}}}"
+        },
+        "outputDataShape": {
+          "kind": "json-schema",
+          "specification": "{\"type\":\"object\",\"properties\":{\"id\":{\"type\":\"integer\",\"format\":\"int64\",\"title\":\"Task ID\",\"description\":\"Unique task identifier\"},\"task\":{\"type\":\"string\",\"title\":\"The task\",\"description\":\"Task line\"},\"completed\":{\"type\":\"integer\",\"title\":\"Task completition status\",\"description\":\"0 - ongoing, 1 - completed\",\"minimum\":0,\"maximum\":1}},\"$schema\":\"http://json-schema.org/schema#\",\"title\":\"Task\"}"
+        },
+        "configuredProperties": {
+          "operationId": "operation-2"
+        }
+      },
+      "tags": [
+        "fetching",
+        "tasks"
+      ],
+      "actionType": "connector",
+      "pattern": "To"
+    },
+    {
+      "id": "io.syndesis:rest-swagger-connector:@syndesis-connectors.version@:_id_:operation-3",
+      "name": "Update task",
+      "description": "Updates task by given identifier",
+      "descriptor": {
+        "camelConnectorGAV": "io.syndesis:rest-swagger-connector:@syndesis-connectors.version@",
+        "camelConnectorPrefix": "swagger-operation",
+        "inputDataShape": {
+          "kind": "json-schema",
+          "specification": "{\"$schema\":\"http://json-schema.org/draft-04/schema#\",\"type\":\"object\",\"properties\":{\"parameters\":{\"type\":\"object\",\"properties\":{\"id\":{\"type\":\"integer\",\"title\":\"id\",\"description\":\"Task identifier\"}}},\"body\":{\"type\":\"object\",\"properties\":{\"id\":{\"type\":\"integer\",\"format\":\"int64\",\"title\":\"Task ID\",\"description\":\"Unique task identifier\"},\"task\":{\"type\":\"string\",\"title\":\"The task\",\"description\":\"Task line\"},\"completed\":{\"type\":\"integer\",\"title\":\"Task completition status\",\"description\":\"0 - ongoing, 1 - completed\",\"minimum\":0,\"maximum\":1}},\"$schema\":\"http://json-schema.org/schema#\",\"title\":\"Task\"}}}"
+        },
+        "outputDataShape": {
+          "kind": "json-schema",
+          "specification": "{\"type\":\"object\",\"properties\":{\"id\":{\"type\":\"integer\",\"format\":\"int64\",\"title\":\"Task ID\",\"description\":\"Unique task identifier\"},\"task\":{\"type\":\"string\",\"title\":\"The task\",\"description\":\"Task line\"},\"completed\":{\"type\":\"integer\",\"title\":\"Task completition status\",\"description\":\"0 - ongoing, 1 - completed\",\"minimum\":0,\"maximum\":1}},\"$schema\":\"http://json-schema.org/schema#\",\"title\":\"Task\"}"
+        },
+        "configuredProperties": {
+          "operationId": "operation-3"
+        }
+      },
+      "tags": [
+        "tasks",
+        "updating"
+      ],
+      "actionType": "connector",
+      "pattern": "To"
+    },
+    {
+      "id": "io.syndesis:rest-swagger-connector:@syndesis-connectors.version@:_id_:operation-4",
+      "name": "Delete task",
+      "description": "Deletes task by given identifier",
+      "descriptor": {
+        "camelConnectorGAV": "io.syndesis:rest-swagger-connector:@syndesis-connectors.version@",
+        "camelConnectorPrefix": "swagger-operation",
+        "inputDataShape": {
+          "kind": "json-schema",
+          "specification": "{\"$schema\":\"http://json-schema.org/draft-04/schema#\",\"type\":\"object\",\"properties\":{\"parameters\":{\"type\":\"object\",\"properties\":{\"id\":{\"type\":\"integer\",\"title\":\"id\",\"description\":\"Task identifier to delete\"}}}}}"
+        },
+        "outputDataShape": {
+          "kind": "none"
+        },
+        "configuredProperties": {
+          "operationId": "operation-4"
+        }
+      },
+      "tags": [
+        "destruction",
+        "tasks"
+      ],
+      "actionType": "connector",
+      "pattern": "To"
+    }
+  ]
+}

